libraries/service_helpers.rb

```diff
@@ -0,0 +1,58 @@
+module CernerSplunk
+  # Mixin Helper methods for the Splunk Service resource
+  module ServiceHelpers
+    include ResourceHelpers
+    require 'win32/service' if RUBY_PLATFORM =~ /mswin|mingw|windows/
+
+    # Reports if the Windows service for the current package exists
+    # @return [Boolean]
+    def service_exists
+      ::Win32::Service.exists?(process_name)
+    end
+
+    def splunk_bin_path
+      Pathname.new(install_dir).join('bin')
+    end
+
+    # Returns the process id of the daemon is running
+    # @return [Integer]
+    def service_pid
+      install_state = node.run_state['splunk_ingredient']['installations'][install_dir]
+      if install_state
+        pid_file = Pathname.new(install_dir).join('var/run/splunk/splunkd.pid')
+        if pid_file.exist?
+          pid = pid_file.readlines.first.to_i || false
+          return pid
+        end
+      end
+      false
+    end
+
+    # Reports if the daemon for the current package is running
+    # @return [Boolean]
+    def service_running
+      service_pid != false
+    end
+
+    # Reports the limits for the current package's running process
+    # @return [String] limits output
+    def service_limits
+      pid = service_pid
+      pid ? Pathname.new("/proc/#{pid}/limits").read : false
+    end
+
+    # Inserts ulimit command into the startup script for the current package's service
+    def try_write_initd_ulimit(ulimit)
```

Comment:
I agree with Josh here. The primary goal of this method is to set the ulimit for the service, the fact that it doesn't do anything if the service isn't setup doesn't have to be reflected in the method name.  That's what the method comment is for.  Otherwise we would be putting try_ in front of a large number of methods we write.